src/pulse_cnt.rs

```diff
@@ -0,0 +1,533 @@
+    NegPos,
+    ///< start from positive value, end to negative value, i.e. +N->-M
+    PosNeg,
+}
+
+impl From<pcnt_unit_zero_cross_mode_t> for PcntZeroCrossMode {
+    fn from(x: pcnt_unit_zero_cross_mode_t) -> Self {
+        match x {
+            esp_idf_sys::pcnt_unit_zero_cross_mode_t_PCNT_UNIT_ZERO_CROSS_POS_ZERO => PcntZeroCrossMode::PosZero,
+            esp_idf_sys::pcnt_unit_zero_cross_mode_t_PCNT_UNIT_ZERO_CROSS_NEG_ZERO => PcntZeroCrossMode::NegZero,
+            esp_idf_sys::pcnt_unit_zero_cross_mode_t_PCNT_UNIT_ZERO_CROSS_NEG_POS => PcntZeroCrossMode::NegPos,
+            esp_idf_sys::pcnt_unit_zero_cross_mode_t_PCNT_UNIT_ZERO_CROSS_POS_NEG => PcntZeroCrossMode::PosNeg,
+            _ => panic!("unknown pcnt_unit_zero_cross_mode_t value {}", x),
+        }
+    }
+}
+
+/// @brief PCNT watch event data
+#[derive(Debug, Copy, Clone)]
+#[allow(dead_code)]
+pub struct PcntWatchEventData {
+    ///< Watch point value that triggered the event
+    pub watch_point_value: i32,
+    ///< Zero cross mode
+    pub zero_cross_mode: PcntZeroCrossMode,
+}
+
+impl From<*const pcnt_watch_event_data_t> for PcntWatchEventData {
+    fn from(x: *const pcnt_watch_event_data_t) -> Self {
+        unsafe {
+            Self {
+                watch_point_value: (*x).watch_point_value,
+                zero_cross_mode: (*x).zero_cross_mode.into(),
+            }
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum PcntEdgeAction {
+    ///< Hold current count value
+    Hold,
+    ///< Increase count value
+    Increase,
+    ///< Decrease count value
+    Decrease,
+}
+
+impl Into<pcnt_channel_edge_action_t> for PcntEdgeAction {
+    fn into(self) -> pcnt_channel_edge_action_t {
+        match self {
+            PcntEdgeAction::Hold => esp_idf_sys::pcnt_channel_edge_action_t_PCNT_CHANNEL_EDGE_ACTION_HOLD,
+            PcntEdgeAction::Increase => esp_idf_sys::pcnt_channel_edge_action_t_PCNT_CHANNEL_EDGE_ACTION_INCREASE,
+            PcntEdgeAction::Decrease => esp_idf_sys::pcnt_channel_edge_action_t_PCNT_CHANNEL_EDGE_ACTION_DECREASE,
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum PcntLevelAction {
+    ///< Keep current count mode
+    Keep,
+    ///< Invert current count mode (increase -> decrease, decrease -> increase)
+    Inverse,
+    ///< Hold current count value
+    Hold,
+}
+
+impl Into<pcnt_channel_level_action_t> for PcntLevelAction {
+    fn into(self) -> pcnt_channel_edge_action_t {
+        match self {
+            PcntLevelAction::Keep => esp_idf_sys::pcnt_channel_level_action_t_PCNT_CHANNEL_LEVEL_ACTION_KEEP,
+            PcntLevelAction::Inverse => esp_idf_sys::pcnt_channel_level_action_t_PCNT_CHANNEL_LEVEL_ACTION_INVERSE,
+            PcntLevelAction::Hold => esp_idf_sys::pcnt_channel_level_action_t_PCNT_CHANNEL_LEVEL_ACTION_HOLD,
+        }
+    }
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+pub struct PcntChanFlags(pcnt_chan_config_t__bindgen_ty_1);
```

Comment:
Nit: you can implement also `pub const new() -> Self` for this struct.
What is the size of this structure? If bigger than - say - 64 bits, perhaps we don't want it to be `Copy`?